src/qibo/hamiltonians/hamiltonians.py

```diff
@@ -525,32 +469,46 @@ def _calculate_dense_from_form(self) -> Hamiltonian:
         matrix = self._get_symbol_matrix(self.form)
         return Hamiltonian(self.nqubits, matrix, backend=self.backend)
 
+    """
     def _calculate_dense_from_terms(self) -> Hamiltonian:
```

Comment:
If I understand correctly, this is unlikely to be relevant anymore, so we could directly drop.